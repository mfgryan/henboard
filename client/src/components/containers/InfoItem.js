// redux dep
import { connect } from "react-redux";
import { removeInfo } from "../../actions/info";
import { closeInfo } from "../../actions/items";
import projects from "../../models/projects.js";
import info from "../../models/info.js";

// component dep
import Info from "../Info";

const mapStateToProps = (state, ownProps) => {
    let project = projects.getCurrentProject(state);
    return {
        project: project.project,
        info: info.getInfoArray(state, project.project, ownProps.name),
        title: ownProps.name
    };
};

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        removeInfo: info => {
            dispatch(
                removeInfo({
                    project: info.project,
                    name: info.name,
                    value: info.value
                })
            );
        },
        closeInfo: (event, title) => {
            dispatch(
                closeInfo({
                    project: ownProps.project,
                    name: title
                })
            );
        }
    };
};

const InfoItem = connect(mapStateToProps, mapDispatchToProps)(Info);

export default InfoItem;
