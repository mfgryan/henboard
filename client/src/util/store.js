// redux dep
import { createStore, applyMiddleware } from "redux";
import { logger, validate } from "../util/middleware.js";
import data from "../util/data.js";
import rootReducer from "../reducers/index.js";

const store = createStore(rootReducer,applyMiddleware(logger,validate));

export { store }
