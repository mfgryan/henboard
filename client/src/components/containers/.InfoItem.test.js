import React from "react";
import InfoItem from "./InfoItem";
import renderer from "react-test-renderer";

// redux dep
import { Provider } from "react-redux";
import { store } from "../../util/store.js";

// router dep
import { BrowserRouter as Router } from "react-router-dom";

it("InfoItem renders correctly", () => {
    const tree = renderer
        .create(
            <Router>
                <Provider store={store}>
                    <InfoItem project="henboard" name="bar" />
                </Provider>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
