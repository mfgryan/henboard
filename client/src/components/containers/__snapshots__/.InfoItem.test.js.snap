exports[`test InfoItem renders correctly 1`] = `
<div
  className="modal"
  draggable="false"
  role="dialog"
  style={
    Object {
      "display": "block",
    }
  }
  tabIndex="-1">
  <div
    className="modal-dialog">
    <div
      className="modal-content"
      role="document">
      <div
        className="modal-header">
        <div
          className="AddItem">
          <h2
            onClick={[Function]}>
            bar
          </h2>
          <div
            className="AddItemInput"
            style={
              Object {
                "visibility": "hidden",
              }
            }>
            <div
              className="form-group">
              <span
                className="input-group input-group-sm">
                <input
                  className="form-control"
                  id={undefined}
                  onChange={[Function]}
                  onKeyPress={[Function]}
                  type="text"
                  value="" />
                <span
                  className="input-group-btn"
                  type="text">
                  <button
                    className="btn btn-default"
                    disabled={false}
                    onClick={[Function]}
                    type="button">
                    Add
                  </button>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="modal-footer">
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button">
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
